File: tests/codegen/snapshots/from_graphql_schema/python/mutation_with_object.py
from enum import Enum
from typing import List

class AddBlogPostsResultAddBlogPostsPosts:
    title: str

class AddBlogPostsResultAddBlogPosts:
    posts: AddBlogPostsResultAddBlogPostsPosts

class AddBlogPostsResult:
    addBlogPosts: AddBlogPostsResultAddBlogPosts

class Color(Enum):
    RED = "RED"
    GREEN = "GREEN"
    BLUE = "BLUE"

class BlogPostInput:
    title: str
    color: Color
    pi: float
    aBool: bool
    anInt: int
    anOptionalInt: int

class AddBlogPostsVariables:
    input: List[BlogPostInput]

File: tests/codegen/snapshots/from_graphql_schema/typescript/mutation_with_object.ts
type AddBlogPostsResultAddBlogPostsPosts = {
    title: string
}

type AddBlogPostsResultAddBlogPosts = {
    posts: AddBlogPostsResultAddBlogPostsPosts
}

type AddBlogPostsResult = {
    addBlogPosts: AddBlogPostsResultAddBlogPosts
}

enum Color {
    RED = "RED",
    GREEN = "GREEN",
    BLUE = "BLUE",
}

type BlogPostInput = {
    title: string
    color: Color
    pi: number
    aBool: boolean
    anInt: number
    anOptionalInt: number
}

type AddBlogPostsVariables = {
    input: BlogPostInput[]
}
